homeassistant/components/slide_local/cover.py

```diff
@@ -0,0 +1,307 @@
+        self._api = api
+        self._id = host
+        self._invert = invert
+        self._name = slide_info.get("device_name", host)
+        self._slide: dict[str, Any] = {}
+        self._slide["pos"] = None
+        self._slide["state"] = None
+        self._slide["online"] = False
+        self._slide["touchgo"] = False
+        self._unique_id = mac
+
+        self.parsedata(slide_info)
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return the device unique id."""
+        return self._unique_id
+
+    @property
+    def name(self) -> str:
+        """Return the device name."""
+        return self._name
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return device specific state attributes."""
+        return {ATTR_ID: self._id, ATTR_TOUCHGO: self._slide["touchgo"]}
+
+    @property
+    def is_opening(self) -> bool:
+        """Return if the cover is opening or not."""
+        return self._slide["state"] == STATE_OPENING
+
+    @property
+    def is_closing(self) -> bool:
+        """Return if the cover is closing or not."""
+        return self._slide["state"] == STATE_CLOSING
+
+    @property
+    def is_closed(self) -> bool:
+        """Return None if status is unknown, True if closed, else False."""
+        return self._slide["state"] == STATE_CLOSED
+
+    @property
+    def available(self) -> bool:
+        """Return False if state is not available."""
+        return self._slide["online"]
+
+    @property
+    def current_cover_position(self) -> int | None:
+        """Return the current position of cover shutter."""
+        pos = self._slide["pos"]
+        if pos is not None:
+            if (1 - pos) <= DEFAULT_OFFSET or pos <= DEFAULT_OFFSET:
+                pos = round(pos)
+            if not self._invert:
+                pos = 1 - pos
+            pos = int(pos * 100)
+        return pos
+
+    async def async_open_cover(self, **kwargs: Any) -> None:
+        """Open the cover."""
+        self._slide["state"] = STATE_OPENING
+        await self._api.slide_open(self._id)
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the cover."""
+        self._slide["state"] = STATE_CLOSING
+        await self._api.slide_close(self._id)
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop the cover."""
+        await self._api.slide_stop(self._id)
+
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
+        """Move the cover to a specific position."""
+        position = kwargs[ATTR_POSITION] / 100
+        if not self._invert:
+            position = 1 - position
+
+        if self._slide["pos"] is not None:
+            if position > self._slide["pos"]:
+                self._slide["state"] = STATE_CLOSING
+            else:
+                self._slide["state"] = STATE_OPENING
+
+        await self._api.slide_set_position(self._id, position)
+
+    async def async_update(self) -> None:
```

Comment:
Also, if we are indeed oging for multiple platforms, I think using a coordinator is the way to go :)

Reply:
Agreed. But for the time being there is only one platform, so a coordinator seeemed not necessary to me. I set it up like tis as you said it would be better to have one config_entry per device. With a coordinator I guess it would make more sense to gather all sides under one config_entry (and then we should rather expand the existing `slide` integration instead of making a new one for the local API).

Reply:
A coordinator is good for centralizing the polling mechanism, and if we have multiple platforms, we might as well just use it so we are not dependent on `async_update` of one entity. I don't really see the point to go 1 config entry -> n devices here because of that

Reply:
Ok. Does it make sense to already add coordinator, even if we only have one platform (in the light of minimizing the PR), or should coordinator be added with second platform?

Reply:
Oh we can add it right away

Reply:
I don't mind if we do